spec/quartz/message_spec.cr

```diff
@@ -0,0 +1,148 @@
+      message._from.not_nil!.tap do |email_address|
+        email_address.addr.should eq email
+        email_address.name.should eq name
+      end
+    end
+  end
+
+  context "destination" do
+    {% for destination in ["to", "cc", "bcc"] %}
+      {% destination = destination.id %}
+      it "{{ destination }} allows setting destinations without name" do
+        message = build
+        message.{{ destination }} email
+        message._{{ destination }}.size.should eq 1
+        message._{{ destination }}.first.addr.should eq email
+      end
+
+      it "{{ destination }} allows setting destinations with name" do
+        message = build
+        message.{{ destination }} name, email
+        message._{{ destination }}.size.should eq 1
+        message._{{ destination }}.first.tap do |email_address|
+          email_address.addr.should eq email
+          email_address.name.should eq name
+        end
+      end
+
+      it "{{ destination }} allows setting destinations by array" do
+        message = build
+        message.{{ destination }} [address, address2]
+
+        message._{{ destination }}.size.should eq 2
+
+        message._{{ destination }}[0].tap do |email_address|
+          email_address.addr.should eq email
+          email_address.name.should eq name
+        end
+
+        message._{{ destination }}[1].tap do |email_address|
+          email_address.addr.should eq email2
+          email_address.name.should eq name2
+        end
+      end
+
+      it "{{ destination }} allows setting destination by named parameter" do
+        message = build
+        message.{{ destination }} name: name, email: email
+        message._{{ destination }}.size.should eq 1
+        message._{{ destination }}.first.tap do |email_address|
+          email_address.addr.should eq email
+          email_address.name.should eq name
+        end
+      end
+    {% end %}
+  end
+
+  context "subject field" do
+    it "allows subject to be set" do
+      message = build
+      message.subject subject
+      message._subject.should eq subject
+    end
+  end
+
+  context "text body field" do
+    it "allows the text body to be set" do
+      message = build
+      message.text text
+      message._text.should eq text
+    end
+
+    it "allows the text body to be set with the #body method" do
+      message = build
+      message.body text
+      message._text.should eq text
+    end
+  end
+
+  context "html body field" do
+    it "allows the html body to be set" do
+      message = build
+      message.html html
+      message._html.should eq html
+    end
+  end
+
+  context "rendering an EMail" do
+    it "renders the email" do
+      message = build
+      message.from address
+      message.to address2
+      message.subject subject
+      message.text text
+      message.html html
+      message.to_email
+
+      # EMail doesn't provide getters for the internal objects,
+      # so we're just going to have to trust it?
```

Comment:
@drujensen any ideas here?

I thought about monkeypatching the EMail class to create some accessors but that feels kind of hacky...

Reply:
yeah, tough one.  I guess we could create a pr `arcage/crystal-email` to add some getters. Not sure what the internal's look like and it thats possible or not.

Reply:
The problem is the same that I ran into with Quartz::Message, the instance variables are the the same as the method names:

```crystal
def to(to_address)
  @to << to_address
end
```

Normally #to would be a getter, but for the sake of the DSL, it's now a setter. I fixed that in 9a895f9 by making the instance variables `@_to` etc and providing a getter but no setter for them.

Reply:
What about making it a setter?
```
def to<< ()
```

Reply:
I'm not opposed to that, but it does change the DSL:

```crystal
class NotificationMailer < ApplicationMailer
  def initialize( @user : User)
    to @user.email.not_nil!
    subject "Notification"
    body render("mailers/notification_email.ecr")
  end
end
```

Would become:

```crystal
class NotificationMailer < ApplicationMailer
  def initialize( @user : User)
    to = @user.email.not_nil!
    subject = "Notification"
    body = render("mailers/notification_email.ecr")
  end
end
```

I'm...kind of divided on that, but I really don't have a strong preference one way or another. I suppose now is as good a time as any to change the DSL if that's what we want to do. 

@amberframework/contributors thoughts? What is your dream Mailer DSL? ✨ 🔨 
```

Reply:
@robacarp Ok.  Lets leave this for now.  We can address it as a separate PR.